consul-lambda/client/ssm.go

```diff
@@ -15,11 +15,18 @@ import (
 // SSMClient provides an API client for interacting with AWS Systems Manager Parameter Store.
 type SSMClient struct {
 	client *ssm.Client
+	tier   types.ParameterTier
 }
 
 // NewSSM creates an instance of the SSMClient from the given AWS SDK config.
-func NewSSM(cfg *aws.Config) *SSMClient {
-	return &SSMClient{client: ssm.NewFromConfig(*cfg)}
+func NewSSM(cfg *aws.Config, aTier bool) *SSMClient {
+	var tier = types.ParameterTierStandard
+
+	if aTier {
+		tier = types.ParameterTierAdvanced
+	}
+
+	return &SSMClient{client: ssm.NewFromConfig(*cfg), tier: tier}
```

Comment:
I had envisioned using a `string` for the tier input. This has two benefits over a `bool` that I can think of:
1. It allows us to be a bit more flexible if the tier types change in the future (for example, if a new tier is added). Please see my [previous suggestion](https://github.com/hashicorp/terraform-aws-consul-lambda/pull/78#discussion_r1253308418) which iterates over the tier types from the AWS SDK and removes any explicit dependency we have on them. If the configured tier type matches, we use it.
1. The new tier config field will get loaded on startup without any need for explicit handling of the field that is [required here](https://github.com/hashicorp/terraform-aws-consul-lambda/pull/78/files#diff-d1c5f26a656a534280b1b54b5ffc39826301e0e23a505cd944eb4372e6d6f290R123-R126). The config is all handled by the `envconfig` package.

W.r.t. to point 1., I completely agree that it is a good idea to include a check for the length of the param. This is a great add! I think we could achieve this in the same if statement:
```go
if c.tier != "" && len(val) > 4096 {
  ...
}
```
Otherwise we always just use the default provided by the AWS SDK (which is standard tier).

What do you think?

Reply:
That sounds perfect, makes a bunch of sense, I'll give it a fix up and re-push